js/fragment_shader.js

```diff
@@ -191,8 +191,26 @@ bool intersectSomething(vec3 origin, vec3 rayDirection, out vec3 intersection,
   return true;
 }
 
+bool intersectSomethingGoingToLight(vec3 origin, vec3 N_aux) {
+  float distToLight = length(lightPosition - origin);
+  vec3 rayDirection = normalize(lightPosition - origin);
+
+  if (dot(N_aux, rayDirection) < 0.0) return false;
```

Comment:
If the V ray is more than 90 degrees from N, it will be negative and we should discard this case. (things get ugly if we don't).